src/collection.cpp

```diff
@@ -624,7 +624,13 @@ void Collection::batch_index(std::vector<index_record>& index_records, std::vect
         if(index_record.indexed.ok()) {
             if(index_record.is_update) {
                 remove_flat_fields(index_record.new_doc);
+                for(auto& field: fields) {
```

Comment:
Instead of looping here, let's see if we can do this processing inside `validate_index_in_memory` which is already looping on the schema fields.